service/src/main/java/cz/fi/muni/pa165/service/MushroomServiceImpl.java

```diff
@@ -21,51 +23,96 @@
     public static final int BASE_YEAR = 1970;
 
     @Inject
-    MushroomDao MushroomDao;
+    private MushroomDao MushroomDao;
 
     @Override
     public void create(Mushroom mushroom) {
-        MushroomDao.create(mushroom);
+
+        try {
+            MushroomDao.create(mushroom);
+        } catch (Exception e) {
+            throw new DataIntegrityViolationException(mushroom + "\n" + e);
```

Comment:
What is the need for wrapping exceptions like this.

Btw even if needed, exceptions usually have a cause parameter in the constructor. It cannot be expected that the user of the service will be aware of this implementation.